compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1607,7 +1607,17 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         let mut err =
             struct_span_err!(self.tcx.sess, ident.span, E0603, "{} `{}` is private", descr, ident);
         err.span_label(ident.span, &format!("private {}", descr));
-        if let Some(span) = ctor_fields_span {
+
+        let mut non_exhaustive = None;
+        // If an ADT is foreign and marked as `non_exhaustive`, then that's
+        // probably why we have the privacy error.
+        // Otherwise, point out if the struct has any private fields.
+        if let Some(def_id) = res.opt_def_id()
+            && !def_id.is_local()
+            && let Some(attr) = self.tcx.get_attr(def_id, sym::non_exhaustive)
+        {
+            non_exhaustive = Some(attr.span);
+        } else if let Some(span) = ctor_fields_span {
             err.span_label(span, "a constructor is private if any of the fields is private");
```

Comment:
Shouldn't it be "are private"? 🤔

Reply:
Both sound fine to me. According to https://english.stackexchange.com/questions/475028/is-any-of-them-used-with-a-singular-or-plural-verb either is correct:

> When any of is followed by a countable plural noun, the verb can be in either singular or plural form, but a singular verb is more common in a formal style

So I won't  change it just to avoid churn.